### PR TITLE
Manually throw error for incomplete expression

### LINKED ISSUE #311: "$" expression outside quote around...
I'm attempting to wrap `<sys/mman.h>` which runs successfully but generators the following line:

```julia
const __ptr_t = $(Expr(:incomplete, "incomplete: premature end of input"))
```

resulting in the error:
```
LoadError: syntax: "$" expression outside quote around /home/colinxs/workspace/crypto/obs/gen/mman.jl:313
```

The corresponding c code is in `<sys/cdefs.h>`:
```c
/* This is not a typedef so `const __ptr_t' does the right thing.  */
#define __ptr_t void *
```

I can blacklist it for now, since it's not used directly in the generated code, but figured I'd file an issue to put it on the radar.

### PR DESCRIPTION
**Pull request by melonedo**
Fix #311

### CODE REVIEW

File: src/generator/macro.jl [hunk 1]
```diff
@@ -296,7 +296,9 @@ function macro_emit!(dag::ExprDAG, node::ExprNode{MacroDefault}, options::Dict)
     end
     str = reduce(add_spaces_for_macros, txts)
     try
-        push!(node.exprs, Expr(:const, Expr(:(=), sym, Meta.parse(str))))
+        ex = Meta.parse(str)
+        Meta.isexpr(ex, :incomplete) && throw(ParseError(ex.args|>only))
```
Comment (Gnimuc) [inline src/generator/macro.jl:300]:
```suggestion
        Meta.isexpr(ex, :incomplete) && throw(Meta.ParseError("failed to parse:  $str"))
```

**Review by Gnimuc — COMMENTED** [review]
To fix #311, we should add support for this macro pattern instead of skipping it.

**Comment by melonedo** [comment]
Can we support this kind of macro? `void *` is tricky even in C...
But at least we must ensure the generated code is valid Julia code, so I forced the generator to ignore this macro.

**Comment by Gnimuc** [comment]
No need to be generic. In this specific case, we can check: if the parsed expr is `incomplete` and the last char in `str` is `*`, then we remove this `*` from `str` and parse again until all of the tailing `*` are recursively removed; now the only thing left should be a valid Julia type, if not, we just give up and skip.

**Comment by Gnimuc** [comment]
@melonedo, is this ready to merge?

**Comment by melonedo** [comment]
I think yes. I do not see any regression locally.

**Comment by Gnimuc** [comment]
Thanks!

**Pull request merged**